index.js

```diff
@@ -107,14 +127,14 @@ NYC.prototype.addContent = function (filename, content) {
   }
 }
 
-NYC.prototype.shouldInstrumentFile = function (relFile, returnImmediately) {
-  // only instrument a file if it's not on the exclude list.
-  for (var i = 0, exclude; (exclude = this.exclude[i]) !== undefined; i++) {
-    if (exclude.test(relFile)) {
-      return false
-    }
+NYC.prototype.shouldInstrumentFile = function (filename, relFile) {
+  relFile = relFile.replace(/^\.\//, '') // remove leading './'.
+
+  if (this.include) {
+    return micromatch.any(filename, this.include) || micromatch.any(relFile, this.include)
```

Comment:
Because `filename` is the complete path, we want to also check the relative-file or globs don't behave as one would expect. Take excluding the test folder:

I would expect to be able to write `test/**`, this however fails to exclude: 
`/Users/benjamincoe/bcoe/yargs/test/validation.js`.

testing against both the absolute and rebased path seems like an elegant solution.